pages/home.py

```diff
@@ -106,20 +113,49 @@ def update_store_inputs(
     inputs[ElementsIDs.MODEL_SELECTION.value] = selected_model
     inputs[ElementsIDs.chart_selected.value] = chart_selected
 
-    return inputs
+    print(f"inputs: {inputs}")
+    url_data = inputs
+    # encode the inputs to be used in the URL
+    url_search = f"?{urlencode(inputs)}"
+    print(f"url_params: {url_data}")
+
+    return inputs, url_search
 
 
 @callback(
     Output(ElementsIDs.INPUT_SECTION.value, "children"),
     Input(ElementsIDs.MODEL_SELECTION.value, "value"),
     Input(ElementsIDs.UNIT_TOGGLE.value, "checked"),
+    Input("url", "search"),
+    State(MyStores.input_data.value, "data"),
 )
-def update_inputs(selected_model, units_selection):
-    # todo here I should first check if some inputs are already stored in the store
+
+# update the inputs based on the model selected, the units selected, and the URL
+def update_inputs(selected_model, units_selection, url_search, stored_data):
     if selected_model is None:
         return no_update
+
     units = UnitSystem.IP.value if units_selection else UnitSystem.SI.value
-    return input_environmental_personal(selected_model, units)
+
+    # Parse URL parameters
+    url_params = parse_qs(url_search.lstrip("?"))
+    url_params = {k: v[0] if len(v) == 1 else v for k, v in url_params.items()}
```

Comment:
this code is very hard to read

Reply:
This line of code is a dictionary derivation that iterates through all the key-value pairs in url_params, extracting the value of the key if it corresponds to a list with only one element to make the change, or making the change to the entire list if the list of values contains more than one element, such as when we follow up with a single set that has more than one value in compare